src/Authenticator/JwtAuthenticator.php

```diff
@@ -0,0 +1,138 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         4.0.0
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Authentication\Authenticator;
+
+use Authentication\Identifier\IdentifierCollection;
+use Authentication\Result;
+use Cake\ORM\Entity;
```

Comment:
@markstory No, that's right. I can't instantiate an entity interface. 😄

The authenticator offers the option to return the JWT token data as it is, without passing it to the identifiers. And the result needs to return an entity class. See the options.

The only problem that remains is how to deal with the "sub" (subject) part of the token. @ADmad  suggested to pass it to an identifier. According to the spec the subject in a token can be a JSON string. I understand it so that this can be in theory any data. Int, string, array... 

I've tried to re-use the OrmIdentifier and Admad suggested the TokenIdentifier. But this requires that the subject is always a single field because then the token identifier could be used to look up the id as if it where a token.

I think we need to find a better way to deal with the data and to transform it into a DB query.